### PR TITLE
feat(profile): add promise support for profileFactory

### PR DESCRIPTION
**Pull request by dannyvaughton**
- feat(createUserProfile) Adding async to functionality to make firebase useful in profileFactory
- docs(Contributing) Adding new troubleshooting section for people struggling to NPM link react-redux-firebase (with two versions of React).

### Description

I realised that simply adding a third parameter "firebase" to profileFactory was not enough to make it useful. It's helpful to have createUserProfile awaiting the results of profileFactory. 

An example use case would be an "invited team user" without an account. On signup, it's useful to check for invited users, merge their profile data & assign with the correct roles & team.

### Check List
If not relevant to pull request, check off as complete

- [x] All tests passing
- [x] Docs updated with any changes or examples if applicable
- [ ] Added tests to ensure new feature(s) work properly

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -84,3 +84,25 @@ It is often convenient to run a local version of `react-redux-firebase` within a
   **What Happened?**
 
   This error most often appears due to Webpack config. A common pattern is to provide `exclude: [/node_modules/]` setting to [babel-loader](https://github.com/babel/babel-loader), which keeps the loader from transforming code within the `node_modules` folder. Now that we have used `npm link`, your project points to your local version of `react-redux-firebase` instead of the one in your `node_modules` folder, so we have to tell the loader to also exclude transforming `react-redux-firebase`.
+
+* `Invalid hook call. Hooks can only be called inside of the body of a function component`:
+
+  **Common Solution**
```
Comment (prescottprue) [inline docs/contributing.md]: Nice work adding this! I have had to deal with this in the past as well. Also, I'll probably add a tidbit about how to do it if you haven't ejected from `react-scripts`

File: src/actions/auth.js [hunk 2]
```diff
@@ -282,7 +287,7 @@ export const createUserProfile = (dispatch, firebase, userData, profile) => {
   if (typeof config.profileFactory === 'function') {
     // catch errors in user provided profileFactory function
     try {
-      profile = config.profileFactory(userData, profile, firebase) // eslint-disable-line no-param-reassign
+      profile = await config.profileFactory(userData, profile, firebase) // eslint-disable-line no-param-reassign
```
Comment (prescottprue) [inline src/actions/auth.js]: Is an error thrown if the profileFactory isn't a promise? If so we can add logic to check if the profileFactory is a promise before running await on it

**Comment by prescottprue** [comment]
Really like the idea, thanks for the PR! Just had a small question about support for a `profileFactory` that isn't a promise.

File: docs/contributing.md (hunk 1 above)
Comment (dannyvaughton) [reply docs/contributing.md]: No worries 👍

File: src/actions/auth.js (hunk 2 above)
Comment (dannyvaughton) [reply src/actions/auth.js]: From my tests, no errors are thrown if the profileFactory is not a promise. I have tried both examples below (and both work with error or warning):

```
// Async
profileFactory: async (userData, profileData, firebase) => {
  const asyncProfileData = await asyncFunctionToPerformOtherActions(profileData, firebase);
  return {
      ...asyncProfileData
  }
}
  
// Sync
profileFactory: (userData, profileData, firebase) => {
  return {
     ...profileData
  }
}
```

I am happy to add a check anyway if you would like?

**Comment by cruzdanilo** [comment]
this piece of code caused the following error in my app:
```
auth.js:1 Uncaught ReferenceError: regeneratorRuntime is not defined
    at auth.js:1
    at Object../node_modules/react-redux-firebase/es/actions/auth.js (auth.js:1)
    at __webpack_require__ (bootstrap:106)
    at Object../node_modules/react-redux-firebase/es/createFirebaseInstance.js (createFirebaseInstance.js:1)
    at __webpack_require__ (bootstrap:106)
    at Object../node_modules/react-redux-firebase/es/ReactReduxFirebaseProvider.js (ReactReduxFirebaseProvider.js:1)
    at __webpack_require__ (bootstrap:106)
    at Module../src/index.jsx (index.jsx:1)
    at __webpack_require__ (bootstrap:106)
    at Object.0 (withTracker.jsx:24)
```
btw, i don't think it's a good idea to add this runtime dependency just for async/await syntactic sugar

**Comment by prescottprue** [comment]
@cruzdanilo Thanks for the input - I had been avoiding it before for that reason, I'll look into changing it

**Pull request merged**